flink-connector/flink-connector-gcp-pubsub/src/main/java/com/google/pubsub/flink/internal/source/reader/PubSubSplitReader.java

```diff
@@ -45,7 +45,10 @@ public PubSubSplitReader(Supplier<NotifyingPullSubscriber> factory) {
   private Multimap<String, PubsubMessage> getMessages() throws Throwable {
     ImmutableListMultimap.Builder<String, PubsubMessage> messages = ImmutableListMultimap.builder();
     for (Map.Entry<String, NotifyingPullSubscriber> entry : subscribers.entrySet()) {
-      entry.getValue().pullMessage().ifPresent(m -> messages.put(entry.getKey(), m));
+      if (entry.getValue().pullMessage() == null) continue;
```

Comment:
`pullMessage` seems to be returning null after switching Optional types. Looking at Guava Optional documentation and `pullMessage`'s implementation, I'm not sure how. This seems related to Subscriber interrupts, since was causing checkpointing failures under high load and the unit test failure below:

```
interrupt_returnsEmptyMessages(com.google.pubsub.flink.internal.source.reader.PubSubSplitReaderTest)  Time elapsed: 0.015 sec  <<< ERROR!
java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.io.IOException: java.lang.NullPointerException
```

Let me know if you have any ideas why null is being returned.

Unrelated to why null is being returned: I removed one of the calls to `entry.getValue().pullMessage()` in the for loop. I'll add the emulator integration test that caught this error in another PR.

Reply:
Ah, it's just because of the test case. For the interrupt_returnsEmptyMessages test in the PubSubSplitReaderTest, we aren't returning an action when pullMessage is called on the mock subscriber. If we add the following to the test, it should resolve the issue:

when(mockSubscriber1.pullMessage()).thenReturn(Optional.absent());
when(mockSubscriber2.pullMessage()).thenReturn(Optional.absent());

Reply:
That fixed it, thanks!

I'll rerun a load test to see if I can replicate the checkpointing issue that I **think** was resolved by adding the null check. Maybe a different local change actually resolved the issue.